libcodechecker/server/client_db_access_handler.py

```diff
@@ -1613,28 +1540,29 @@ def addFileContent(self, content_hash, content, encoding):
             return False
         return True
 
-    def __is_same_event_path(self, start_bugevent_id, events):
+    def __is_same_event_path(self, report_id, events):
         """
         Checks if the given event path is the same as the one in the
         events argument.
         """
         try:
-            # There should be at least one bug event.
-            point2 = self.__session.query(BugPathEvent).get(start_bugevent_id)
+            q = self.__session.query(BugPathEvent) \
+                .filter(BugPathEvent.report_id == report_id) \
+                .order_by(BugPathEvent.order)
+
+            for i, point2 in enumerate(q):
+                if i == len(events):
```

Comment:
Do we want to compute the len of events in every iteration?
Are you sure this condition is correct? Especially in the last iteration?